test/MUIDataTable.test.js

```diff
@@ -429,6 +430,24 @@ describe('<MUIDataTable />', function() {
     assert.deepEqual(state.filterList, [['Joe James'], [], [], [], []]);
   });
 
+  it('should apply columns prop change for filterList', () => {
+    const mountShallowWrapper = mount(shallow(<MUIDataTable columns={columns} data={data} />).get(0));
+    const instance = mountShallowWrapper.instance();
+    instance.initializeTable(mountShallowWrapper.props());
+    // now use updated columns props
+    const newColumns = cloneDeep(columns);
+    newColumns[0].options.filterList = ['Joe James'];
+    mountShallowWrapper.setProps({ columns: newColumns });
+    mountShallowWrapper.update();
+    // mimic componentDidUpdate(prevProps) {
```

Comment:
I think we can remove these two comments. My reasoning is that they will be brittle; if the actual code changes, the comments will be out of date, and could lead to confusion.

Reply:
Sounds good to me.